plugin.rb

```diff
@@ -81,33 +82,33 @@ class Engine < ::Rails::Engine
     reactions << like_reaction
   end
 
-  add_to_serializer(:post, :current_user_reactions) do
-    return [] unless scope.user.present?
-    reactions = object.reactions.map do |reaction|
+  add_to_serializer(:post, :current_user_reaction) do
+    return nil unless scope.user.present?
+    object.reactions.each do |reaction|
       reaction_user = reaction.reaction_users.find { |ru| ru.user_id == scope.user.id }
 
       next unless reaction_user
 
-      {
+      return {
         id: reaction.reaction_value,
         type: reaction.reaction_type.to_sym,
         can_undo: reaction_user.can_undo?
-      }
-    end.compact
+      } if reaction.reaction_users_count
+    end
 
     like = object.post_actions.find do |l|
```

Comment:
we could just have this as the last statement here, if it's not nil we will return this, otherwise nil

Reply:
here we are using `each` loop, we don't need to return any value. I am returning the value as we have only one reaction per user and if we find any reaction we directly return this value to withdraw control from this method.